scripts/aws/cloudformation-infra.yaml

```diff
@@ -10,6 +10,20 @@ Parameters:
     Description: Second byte from the left for VPC CIDR
 
 Resources:
+  AttachmentBucket:
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Retain
+    Properties:
+      ObjectLockEnabled: true
```

Comment:
**[Approfondimento]**
Dalla [doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html#cfn-s3-bucket-objectlockconfiguration) _sembra_ che ``ObjectLockEnabled`` voglia ``ObjectLockConfiguration``.

Non son certo, male che vada con questa versione hai una idea di cosa aggiungere.

**Nota:** hai idea di quali siano i default chiedendo solo di abilitare ``ObjectLock``? Utile documentarli. Parlo di retention etc.

Reply:
Anche a me dalla documentazione sembrava utile, in realtà non serve. 
Ho provato il template **(aggiungo una nota)**

Effetti: nessuna retention di default sui nuovi oggetti e nessun "legal hold". Ma possibilità di mettere legal hold e retention sui singoli oggetti.

Reply:
OK. Quindi lasciando solo ``ObjectLockEnabled`` che cosa si ottiene?
Alcune opzioni, dimmi tu quella che è vera:
1. nulla, dipende poi quando andiamo a fare gli upload dei singoli oggetti
2. per default tutti gli oggetti sotto quel bucket non sono cancellabili coi normali permessi
3. ?

Fammi sapere così possiamo chiudere questo commento.

Reply:
1